File: src/main/java/org/kane/domain/service/weight_record/WeightRecordService.java
package org.kane.domain.service.weight_record;

import org.kane.domain.DTO.entityDTO.weight_record.CurrentWeightRecordShowDTO;
import org.kane.domain.DTO.entityDTO.weight_record.WeightRecordCreateDTO;
import org.kane.domain.DTO.entityDTO.weight_record.WeightRecordShowDTO;
import org.kane.domain.DTO.entityDTO.weight_record.for_chart.WeightChartDataDTO;
import org.kane.domain.DTO.request.WeightChartRequest;

import java.security.Principal;
import java.util.List;

public interface WeightRecordService {
    List<WeightRecordShowDTO> getAllRecords(Principal principal);

    CurrentWeightRecordShowDTO getCurrentRecord(Principal principal);

    CurrentWeightRecordShowDTO createRecord(Principal principal, WeightRecordCreateDTO weightRecordCreateDTO);

    WeightChartDataDTO formChart(Principal principal, WeightChartRequest weightChartRequest);
}

File: src/app/service/diary/weight-record.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Endpoint} from '../../util/endpoint';
import {Observable} from 'rxjs';
import WeightRecordShowDTO from '../../DTO/entity_dto/diary/weight_record/weight-record-show.dto';
import CurrentWeightRecordShowDTO from '../../DTO/entity_dto/diary/weight_record/current-weight-record-show.dto';
import WeightRecordCreateDTO from '../../DTO/entity_dto/diary/weight_record/weight-record-create.dto';
import WeightChartRequest from '../../DTO/requests/weight-chart.request';
import WeightChartDataDTO from '../../DTO/entity_dto/diary/weight_record/weight-chart-data.dto';

@Injectable({
  providedIn: 'root',
})
export default class WeightRecordService {
  private readonly http = inject(HttpClient);
  private readonly recordAPI = new Endpoint('weightRecord');

  getAllWeightRecords():Observable<WeightRecordShowDTO[]>{
    return this.http.get<WeightRecordShowDTO[]>(
      this.recordAPI.builder()
        .points("all")
        .build())
  }

  getCurrentWeightRecord():Observable<CurrentWeightRecordShowDTO>{
    return this.http.get<CurrentWeightRecordShowDTO>(
      this.recordAPI.builder()
        .points("current")
        .build());
  }

  createRecord(record: WeightRecordCreateDTO): Observable<CurrentWeightRecordShowDTO> {
    return this.http.put<CurrentWeightRecordShowDTO>(
      this.recordAPI.builder()
        .points("create")
        .build(),
      record);
  }

  createChart(record: WeightChartRequest):Observable<WeightChartDataDTO>{
    return this.http.post<WeightChartDataDTO>(
      this.recordAPI.builder()
        .points("chart")
        .build(),
      record);
  }
}
